### PR TITLE
Proposed change of hgv_time_restrictions field to Time Domain from GDF 5.1

### PR DESCRIPTION
**Pull request by roman-ianivskyy**
… and shorter format of Time Domain from GDF 5.1

- [ ] ~Update tests~
- [x] Update docs

### CODE REVIEW

**Comment by roman-ianivskyy** [comment]
Hello, @nvkelso, could you please check this proposed change to time format for HGV restrictions? I assume it was not yet included in Tilezen datasources. Or if it was already released - what is the process for such breaking changes?

File: docs/layers.md [hunk 1]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `hnn` - Hour. Defines either a particular hour (0 to 23) within a particular day (if previously defined) in a starting date; or a duration of nn hours, i.e., nn*60 minutes. For example, [(y1991M11d14h5m30s19) {h10}] means from 14 November 1991, 5:30:19 a.m. to 14 November 1991 at 3:30:19 p.m. Notice that {h1} = {m60}
+* `mnn` - Minute. Defines either a particular minute (0 to 59) within a particular hour (if previously defined) in a starting date; or a duration of nn minutes, i.e., nn*60 seconds. For example, [(y1991M11d14h5m30s19) {m11}] means from 14 November 1991 at 5:30:19 a.m. to 14 November 1991 at 5:41:19 a.m. Notice that {m1} = {s60}
+* `snn` - Second. Defines either a particular second (0 to 59) within a particular minute (if previously defined) in a starting date; or a duration of nn seconds. For example, [(y1991M11d14h5m30s19) {s21}] means from 14 November 1991 at 5:30:19 a.m. to 14 November 1991 at 5:30:40 a.m. Notice that {m1} = {s60}
+* `znn - Fuzzy symbols for special time terms. Currently only 2 variants are used: [(z1){z51}] means dawn til dusk (nighttime) and [(z2){z52}] means dusk til dawn (daytime). May be extended in future
```
Comment (nvkelso) [inline docs/layers.md:1613]: To: `znn` (add the closing tick)

File: docs/layers.md [hunk 2]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+
+Starting Dates and time intervals are defined by means of a set of graphical symbols allowing the description of years, months, weeks, days, and so on down to the smallest time unit, which is the second. The symbols have to be organized in a sequential order starting with the longest time unit. Attached to a starting date, the interval constitutes a basic Time Domain. Without a starting date, it just indicates a duration. If the very first time type code in duration section is preceded by a minus sign, it means that the duration is counted in the reverse order. The following list describes starting date and time interval unit types:
+* `ynn` - Year. Defines either a particular nnnn year in a starting date; or a duration of nn years. For example, [(y1991M11d14h5m30s19){y1}] means from 14 November 1991, 5:30:19 a.m. to 14 November 1992, 5:30:19 a.m. If there is no identical calendar date in the year in question, which occurs only for 29 February, “plus 1 year” leads to 28 February of the following year. Notice that {y1 = {M12}
```
Comment (nvkelso) [inline docs/layers.md:1603]: Nit 1: add whitespace newline return before starting a list.

Nit 2: any example code should be quoted, like `[(y1991M11d14h5m30s19){y1}]`

Nit 3: `{y1 = {M12}` to `{y1} = {M12}` (add missing `}` brace?

File: docs/layers.md [hunk 3]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `Mnn` - Month. Defines either a particular month (1 to 12) within a particular year, or any year when no `y` information is given in starting date; or a duration of nn months. For example, [(y1991M11d14h5m30s19) {M3}] means from 14 November 1991, 5:31:19 a.m. to 14 February 1992, 5:30:19 a.m.
+* `wnn` - Week. Defines either a week (1 to 53) within a previously defined year, or any year when no `y` information is given in a starting date; or a duration of nn weeks, i.e., nn*7 days. For example, [(y1991M11d14h5m30s19) {w2}] means from 14 November 1991 at 5:30:19 a.m. to 28 November 1991, 5:30:19 a.m. Notice that {w1} = {d7}
```
Comment (nvkelso) [inline docs/layers.md:1605]: Quote code snippets:

- `[(y1991M11d14h5m30s19) {w2}]`
- `{w1}` = `{d7}`

File: docs/layers.md [hunk 4]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `dnn` - Day. Defines either a particular day (1 to 28, 29, 30, 31) within a particular month if previously defined with the `M` format in a starting date; or a duration of nn days, i.e., nn*24 hours. For example, [(y1991M11d14h5m30s19) {d2}] means from 14 November 1991 at 5:30:19 a.m. to 16 November 1991, 5:30:19 a.m. Notice that {d1} = {h24}
```
Comment (nvkelso) [inline docs/layers.md:1606]: Quote code: `[(y1991M11d14h5m30s19) {d2}]`

File: docs/layers.md [hunk 5]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `tn` - Day. Only in starting date defines a particular weekday (1:Sunday, 2:Monday, 3:Tuesday, 4:Wednesday, 5:Thursday, 6:Friday, 7:Saturday) in a previously (if any) defined week. For example, (M5t2) means each Monday in the fifth month (May) of any year, at 0:00:00 a.m.
+* `fxn` - Day. Only in starting date defines a particular weekday in a previously (if any) defined month, with the following rules: n is used as in the “t” format with the same domain of values, 1:Sunday up to 7:Saturday. For “x” one of the following values has to be substituted: 1:first, 2:second, 3:third, 4:fourth, 5:fifth. For example: (...f12) means the first Monday at 0:00:00 a.m.
```
Comment (nvkelso) [inline docs/layers.md:1608]: I don't follow this example

File: docs/layers.md [hunk 6]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `lxn` - Day. Only in starting date defines a particular weekday in a previously (if any) defined month, with the following rules: n is used as for the “t” format with the same domain of values, 1:Sunday up to 7:Saturday. x has to be chosen from the following set: 1:first, 2:second, 3:third, 4:fourth, 5:fifth. Values are given in reverse order starting from the end of the month and have the meaning: 1=last, 2=last but one, 3= last but two, etc. For example, (...l12) means the last Monday at 0:00:00 a.m.
```
Comment (nvkelso) [inline docs/layers.md:1609]: Quote the code snippets

File: docs/layers.md [hunk 7]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `hnn` - Hour. Defines either a particular hour (0 to 23) within a particular day (if previously defined) in a starting date; or a duration of nn hours, i.e., nn*60 minutes. For example, [(y1991M11d14h5m30s19) {h10}] means from 14 November 1991, 5:30:19 a.m. to 14 November 1991 at 3:30:19 p.m. Notice that {h1} = {m60}
```
Comment (nvkelso) [inline docs/layers.md:1610]: Quote code snippets

File: docs/layers.md [hunk 8]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `mnn` - Minute. Defines either a particular minute (0 to 59) within a particular hour (if previously defined) in a starting date; or a duration of nn minutes, i.e., nn*60 seconds. For example, [(y1991M11d14h5m30s19) {m11}] means from 14 November 1991 at 5:30:19 a.m. to 14 November 1991 at 5:41:19 a.m. Notice that {m1} = {s60}
```
Comment (nvkelso) [inline docs/layers.md:1611]: Quote code snippets

File: docs/layers.md [hunk 9]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `snn` - Second. Defines either a particular second (0 to 59) within a particular minute (if previously defined) in a starting date; or a duration of nn seconds. For example, [(y1991M11d14h5m30s19) {s21}] means from 14 November 1991 at 5:30:19 a.m. to 14 November 1991 at 5:30:40 a.m. Notice that {m1} = {s60}
```
Comment (nvkelso) [inline docs/layers.md:1612]: Quote code snippets

File: docs/layers.md (hunk 1 above)
Comment (nvkelso) [reply docs/layers.md:1613]: Quote all the inline code snippets, please

File: docs/layers.md [hunk 10]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `znn - Fuzzy symbols for special time terms. Currently only 2 variants are used: [(z1){z51}] means dawn til dusk (nighttime) and [(z2){z52}] means dusk til dawn (daytime). May be extended in future
+* `string_value` - Externally specified string value. For example [*EASTER(h8){h14}] means from 8:00 am to 10:00 pm during Easter
```
Comment (nvkelso) [inline docs/layers.md:1614]: Quote code snippets

File: docs/layers.md [hunk 11]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+
+Starting Dates which are composed of several time units (e.g., 14 November 1991) are defined by placing the symbols sequentially in a hierarchical order.
+
+Time periods (duration) which are composed of several time interval units are represented by listing the individualsymbols sequentially in hierarchical order. The total time duration of a particular combination of symbols is the sum of all single time intervals. For example, {y2M1w2} means a period of 2 years, 1 month, and 2 weeks. Since Time Domains can be considered as a set of the smallest time unit described here, the second, Time Domains may also be combined with set operations, such as:
+* Union of sets notation: `+`
```
Comment (nvkelso) [inline docs/layers.md:1619]: Nit, newline before list in markdown makes all the markdown renderers happier

**Review by nvkelso — CHANGES_REQUESTED** [review]
Please address the nits in the documentation and then I'll approve.

Tilezen uses documents our implementation of SemVer in 

- https://github.com/tilezen/vector-datasource/blob/master/docs/SEMANTIC-VERSIONING.md#patch-version-increments

Since this is an optional roads layer property, it isn't explicitly covered in the contract (which is mostly focused on layers, kinds, and min_zoom changes. Adding optional properties is in minor version. Modifying the value definitions (since this is really a large reclassification) would fall into MINOR or PATCH territory. But we'll tag it in the upcoming v1.9 release and can merge this PR when you're ready with the changes.

We've never added support for it in the mainline vector-datasource repo here, which also gives some more flexibility.

File: docs/layers.md (hunk 1 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: thanks, fixed quotes

File: docs/layers.md (hunk 2 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md [hunk 12]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `wnn` - Week. Defines either a week (1 to 53) within a previously defined year, or any year when no `y` information is given in a starting date; or a duration of nn weeks, i.e., nn*7 days. For example, [(y1991M11d14h5m30s19) {w2}] means from 14 November 1991 at 5:30:19 a.m. to 28 November 1991, 5:30:19 a.m. Notice that {w1} = {d7}
+* `dnn` - Day. Defines either a particular day (1 to 28, 29, 30, 31) within a particular month if previously defined with the `M` format in a starting date; or a duration of nn days, i.e., nn*24 hours. For example, [(y1991M11d14h5m30s19) {d2}] means from 14 November 1991 at 5:30:19 a.m. to 16 November 1991, 5:30:19 a.m. Notice that {d1} = {h24}
+* `tn` - Day. Only in starting date defines a particular weekday (1:Sunday, 2:Monday, 3:Tuesday, 4:Wednesday, 5:Thursday, 6:Friday, 7:Saturday) in a previously (if any) defined week. For example, (M5t2) means each Monday in the fifth month (May) of any year, at 0:00:00 a.m.
```
[the remark this answers is not in the document]
Comment (roman-ianivskyy) [reply docs/layers.md]: verified with document and implementation - it really starts with 1:Sunday

File: docs/layers.md (hunk 5 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fxn:  x=1 - first, n=2 - Monday, so f12 means first Monday
 f22 would mean second Monday

File: docs/layers.md (hunk 6 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md (hunk 7 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md (hunk 8 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md (hunk 9 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md (hunk 10 above)
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md [hunk 13]
```diff
@@ -1595,34 +1595,34 @@ The `hgv` property indicates general truck heavy goods vehicle truck access, val
...
+* `znn - Fuzzy symbols for special time terms. Currently only 2 variants are used: [(z1){z51}] means dawn til dusk (nighttime) and [(z2){z52}] means dusk til dawn (daytime). May be extended in future
+* `string_value` - Externally specified string value. For example [*EASTER(h8){h14}] means from 8:00 am to 10:00 pm during Easter
+
+Starting Dates which are composed of several time units (e.g., 14 November 1991) are defined by placing the symbols sequentially in a hierarchical order.
+
+Time periods (duration) which are composed of several time interval units are represented by listing the individualsymbols sequentially in hierarchical order. The total time duration of a particular combination of symbols is the sum of all single time intervals. For example, {y2M1w2} means a period of 2 years, 1 month, and 2 weeks. Since Time Domains can be considered as a set of the smallest time unit described here, the second, Time Domains may also be combined with set operations, such as:
```
[the remark this answers is not in the document]
Comment (roman-ianivskyy) [reply docs/layers.md]: fixed

File: docs/layers.md (hunk 11 above)
Comment (roman-ianivskyy) [reply docs/layers.md:1621]: fixed

File: docs/layers.md [hunk 14]
```diff
@@ -1606,8 +1606,8 @@ Starting Dates and time intervals are defined by means of a set of graphical sym
 * `wnn` - Week. Defines either a week (1 to 53) within a previously defined year, or any year when no `y` information is given in a starting date; or a duration of nn weeks, i.e., nn*7 days. For example, `[(y1991M11d14h5m30s19) {w2}]` means from 14 November 1991 at 5:30:19 a.m. to 28 November 1991, 5:30:19 a.m. Notice that `{w1}` = `{d7}`
```
Comment (nvkelso) [inline docs/layers.md:1606]: Should this be `1 to 52` (not 53) weeks in a year?

File: docs/layers.md (hunk 5 above)
Comment (nvkelso) [reply docs/layers.md]: I understand now, thanks

**Review by nvkelso — APPROVED** [review]
Left one nit about number of weeks in a year.

File: docs/layers.md (hunk 14 above)
Comment (nvkelso) [reply docs/layers.md:1606]: Wow, 53 is a thing: https://nrf.com/resources/4-5-4-calendar

**Pull request merged**